### PR TITLE
feat(kmodal): style redesign [khcp-1897]

### PR DESCRIPTION
**Pull request by kaiarrowood**
### Summary
Update KModal with new design.
For khcp-1897.

#### Changes made:
* Default `KButton` `appearance` property to `outline`
* Remove default values for modal `title` and `content`
* Make `title` required for accessibility
* Add `hideTitle` prop if they don't want the title displayed
* Remove `helpText` prop and `help` slot
* Swap order of action and cancel buttons (cancel comes first)
* Add `action-buttons` slot
* Update style classes to prefix with `k-` (class without `k-` left in for backwards compatibility)
* Update snaps

Figma: https://www.figma.com/file/sfSZhMfpzCzddEhd8tdk9s/Kong_Design-library?node-id=209%3A1086

image

### PR Checklist
- [x] Does not introduce dependencies
- [x] **Functional:** all changes do not break existing APIs and if so, bump major version.
- [x] **Tests pass:** check the output of yarn test packages/<Kongponent>
- [x] **Naming:** the files and the method and prop variables use the same naming conventions as other Kongponents
- [x] **Framework style:** abides by the essential rules in Vue's style guide
- [x] **Cleanliness:** does not have formatting issues, unused code (e.g., console.logs), or leftover comments
- [x] **Docs:** includes a technically accurate README, uses JSDOC where appropriate

### CODE REVIEW

File: packages/KModal/KModal.vue [hunk 1]
```diff
@@ -5,39 +5,43 @@
...
+      class="k-modal-backdrop modal-backdrop"
       @click="close">
 
       <div
-        class="modal-dialog"
+        class="k-modal-dialog modal-dialog"
         @click.stop>
-        <div class="modal-content">
+        <div class="k-modal-content modal-content">
           <div
```
Comment (chrisvdp) [inline packages/KModal/KModal.vue:15]: This should be a heading tag.

File: packages/KModal/KModal.vue [hunk 2]
```diff
@@ -5,39 +5,43 @@
     class="k-modal"
```
Comment (chrisvdp) [inline packages/KModal/KModal.vue:5]: We should add an `aria-labelledby` and point it to the title

File: packages/KModal/KModal.vue [hunk 3]
```diff
@@ -5,39 +5,43 @@
...
           </div>
-          <div class="modal-footer">
+          <div class="k-modal-footer modal-footer d-flex">
             <slot name="footer-content">
-              <KButton
-                :appearance="actionButtonAppearance"
-                @click="proceed">
-                {{ actionButtonText }}
-              </KButton>
               <KButton
```
Comment (chrisvdp) [inline packages/KModal/KModal.vue:31]: this should have a keyboard event, like onEnter

File: packages/KModal/KModal.vue [hunk 4]
```diff
@@ -5,39 +5,43 @@
...
                 :appearance="cancelButtonAppearance"
                 @click="close">
                 {{ cancelButtonText }}
               </KButton>
+              <div class="k-modal-action-buttons">
+                <slot name="action-buttons">
+                  <KButton
```
Comment (chrisvdp) [inline packages/KModal/KModal.vue:38]: this should have a keyboard event, like onEnter

**Review by chrisvdp — COMMENTED** [review]
When the modal is set to `isVisible true` we should also set the focus to the first focusable element. In this case that's the close button.

File: packages/KModal/KModal.vue (hunk 1 above)
Comment (kaiarrowood) [reply packages/KModal/KModal.vue:15]: I don't really want to mess with styles associated with headings. Is there a `role` I can use?

File: packages/KModal/KModal.vue (hunk 4 above)
Comment (kaiarrowood) [reply packages/KModal/KModal.vue:41]: Using `@keyup.enter`, love the idea thanks!

File: packages/KModal/KModal.vue (hunk 3 above)
Comment (kaiarrowood) [reply packages/KModal/KModal.vue:33]: Using `@keyup.esc`, thanks for the idea!!

File: packages/KModal/KModal.vue (hunk 2 above)
Comment (kaiarrowood) [reply packages/KModal/KModal.vue:6]: Now using `aria-modal`, but `aria-labelledby` requires an `id`? Instead using `aria-label` with the `title` prop.... if they instead use the `header-content` slot with some sort of complex HTML, this won't work though, and wouldn't work for an `aria-labelledby` would it? Thinking of making `title` possibly a required prop?

File: packages/KModal/KModal.vue (hunk 1 above)
Comment (kaiarrowood) [reply packages/KModal/KModal.vue:17]: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/heading_role will work?

File: packages/KModal/KModal.vue (hunk 2 above)
Comment (kaiarrowood) [reply packages/KModal/KModal.vue:6]: I have made the `title` prop required. It is always used for the `aria-label`. Added a new prop `hideTitle` if they don't want to display it for some reason.

File: docs/components/modal.md [hunk 5]
```diff
@@ -39,41 +43,48 @@ export default {
...
-- `helpText` - Help text to display below header if not using slot
+- `title` - Text displayed in header if not using slot (**Note**: this field is still required for accessibility reasons even if using the slot)
+- `hideTitle` - If not using the `header-content` slot, tells the component whether or not to display the title
 - `content` - Text to display in content if not using slot
 
 **Buttons & Appearance**
 - `actionButtonText` - Change the text content of the submit/proceed button
 - `actionButtonAppearance` - Change the [appearance](/components/button.html#props) of the submit/proceed
 - `cancelButtonText` - Change the text content of the close/cancel button
+- `cancelButtonAppearance` - Change the [appearance](/components/button.html#props) of the cancel
```
Comment (dustinryerson) [inline docs/components/modal.md:54]:
```suggestion
- `cancelButtonAppearance` - Change the [appearance](/components/button.html#props) of the cancel button
```

File: docs/components/modal.md [hunk 6]
```diff
@@ -39,41 +43,48 @@ export default {
...
-- `cancelButtonAppearance` - Change the [appearance](/components/button.html#props) of the close/cancel
-- `title` - Text to display in header if not using slot
-- `helpText` - Help text to display below header if not using slot
+- `title` - Text displayed in header if not using slot (**Note**: this field is still required for accessibility reasons even if using the slot)
+- `hideTitle` - If not using the `header-content` slot, tells the component whether or not to display the title
 - `content` - Text to display in content if not using slot
 
 **Buttons & Appearance**
 - `actionButtonText` - Change the text content of the submit/proceed button
 - `actionButtonAppearance` - Change the [appearance](/components/button.html#props) of the submit/proceed
```
Comment (dustinryerson) [inline docs/components/modal.md:52]:
```suggestion
- `actionButtonAppearance` - Change the [appearance](/components/button.html#props) of the submit/proceed button
```

**Review by dustinryerson — APPROVED** [review]
LGTM, great work! 🎉

**Pull request merged**